### ISSUE TITLE
Plain Java support

### ISSUE THREAD

**Issue by dominickleppich** [sevdokimov/log-viewer#218]
Hi,

this project looks really promising for my needs and I would like to thank you for your work.

I am working on a JSF web application and would like to embed this log viewer into it. The documentation on this topic is not written yet and the source code seems to be tightly coupled to Spring code. Am I missing something or is there no support for plain Java web projects right now?

Thanks in advance :)

**Comment by sevdokimov** [comment]
You can use log viewer in a plain java application, you have to register `com.logviewer.web.LogViewerServlet` as a servlet and create a default spring context. 
See https://github.com/sevdokimov/log-viewer/tree/master/demo/simple-java-web-application/ as a sample project. 
There is [MyLogViewerServlet.java](https://github.com/sevdokimov/log-viewer/blob/master/demo/simple-java-web-application/src/main/java/com/logviewer/demo/MyLogViewerServlet.java) class that extends `com.logviewer.web.LogViewerServlet` and creates a default spring context in `init()` method.

**Comment by dominickleppich** [comment]
Hi, thanks for the answer. I already found this demo sample, but it seems to only work with a Spring context. The web application I'm working on is not a Spring application, but a JSF without Spring application. 

Can this Spring context be used in a non-Spring application without any issues?

**Comment by sevdokimov** [comment]
yes, that spring context will be used by log-viewer only, other parts of the application may be non-spring
